### PR TITLE
Query: Add pop-up when Explain Checkbox is disabled

### LINKED ISSUE #6632: Explain checkbox is unavailable for any actions
### Is your proposal related to a problem?

The issue is in the thanos-query component , the Explain checkbox is unavailable when we use Prometheus as an engine .

### Describe the solution you'd like

The Explain checkbox should be available whenever we execute a query.

### Describe alternatives you've considered

Or a pop-up alert should be provided explaining the reason or any message of its unavailability.

### Additional context

N/A

### PR DESCRIPTION
**Pull request by lmarques03**
Signed-off-by: Luís Marques [henriquecondado@gmail.com](mailto:henriquecondado@gmail.com)

* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

Fixes #6632 

## Verification

image
image
image
image

### CODE REVIEW

**Review by MichaHoffmann — CHANGES_REQUESTED** [review]
Couldnt we just hide this button if the prometheus engine is selected?

**Comment by lmarques03** [comment]
> Couldnt we just hide this button if the prometheus engine is selected?

I can do that if you prefer it that way :)

**Comment by saswatamcode** [comment]
> Couldnt we just hide this button if the prometheus engine is selected?

Well yeah, but I'd like users to know that the feature exists, even if they default to `prometheus` engine. So disabling instead of hiding completely would be a way to do that. That way they'd be more proactive in trying it out. Wdyt @MichaHoffmann? 🙂

File: pkg/store/bucket.go [hunk 1]
```diff
@@ -2864,14 +2864,13 @@ func (r *bucketIndexReader) decodeCachedPostings(b []byte) (index.Postings, []fu
 	)
 	if isDiffVarintSnappyEncodedPostings(b) || isDiffVarintSnappyStreamedEncodedPostings(b) {
 		s := time.Now()
-		clPostings, err := decodePostings(b)
+		l, err = decodePostings(b)
```
Comment (saswatamcode) [inline pkg/store/bucket.go:2867]: I think the order of commits is a bit strange here. This change is unrelated to this PR.

File: pkg/ui/react-app/src/components/Checkbox.tsx [hunk 2]
```diff
@@ -3,14 +3,18 @@ import { FormGroup, Label, Input, InputProps } from 'reactstrap';
...
-const Checkbox: FC<CheckboxProps> = ({ children, wrapperStyles, id, ...rest }) => {
+const Checkbox: FC<CheckboxProps> = ({ children, wrapperStyles, id, disabled, isExplainCheckbox, ...rest }) => {
   return (
     <FormGroup className="custom-control custom-checkbox" style={wrapperStyles}>
-      <Input {...rest} id={id} type="checkbox" className="custom-control-input" />
+      <Input {...rest} id={id} type="checkbox" className="custom-control-input" disabled={disabled} />
       <Label style={{ userSelect: 'none' }} className="custom-control-label" for={id}>
         {children}
+        {isExplainCheckbox && disabled && (
+          <div className="popup-message">This functionality is only available when using the Thanos engine</div>
```
Comment (saswatamcode) [inline pkg/ui/react-app/src/components/Checkbox.tsx:16]: Let's explain the feature too.
```suggestion
          <div className="popup-message">Explain allows you to view an expandable query plan of the PromQL query similar to SQL EXPLAIN. This functionality is only available when using the Thanos engine</div>
```

**Review by saswatamcode — COMMENTED** [review]
Generally LGTM but you might want to fix the commits, so only your changes exist on the branch. 🙂

**Comment by MichaHoffmann** [comment]
> > Couldnt we just hide this button if the prometheus engine is selected?
> 
> Well yeah, but I'd like users to know that the feature exists, even if they default to `prometheus` engine. So disabling instead of hiding completely would be a way to do that. That way they'd be more proactive in trying it out. Wdyt @MichaHoffmann? 🙂

Okay fair enough! Lets proceed with the tooltip!

**Review by saswatamcode — APPROVED** [review]
LGTM!

**Pull request merged**